### ISSUE TITLE
Area2D stops working after resizing a Shape

### ISSUE THREAD

**Issue by Dr4kzor** [godotengine/godot#33369]
**Godot version:**
3.1.1

**OS/device including version:**
Windows 7 Ultimate

**Issue description:**
As soon as a Area2D Shape changes its size this Area2D stops sending signals

**Steps to reproduce:**
Create an Area2D, add a CollisionShape2D, set Shape to RectangleShape2D extents (10,10)
Add an AnimationPlayer and make the extents value go from (10,10) to (1000,1000) on the animation, make it auto play and loop for simplicity
Create another Area2D, add any CollisionShape and make sure the growing area will overlap this after some growth.
Make sure the Layer and mask are set to 1 on both areas
Connect the signals for the first Area and check that it never triggers
Activate the debug "visible collision shapes"

**Minimal reproduction project:**
[BugTest.zip](https://github.com/godotengine/godot/files/3810975/BugTest.zip)

**Comment by KoBeWi** [comment]
Is it maybe related to #19271?

**Comment by Dr4kzor** [comment]
I guess its not the same as this will not trigger any signal. If its already inside a Area2D it will trigger area_exited only once and after this the Area2D will no longer trigger anything.

**Comment by KoBeWi** [comment]
Ok, this might have something to do with the fact that you are using animation. Animating some properties related to physics etc. tends to cause problems.

**Comment by Dr4kzor** [comment]
My original code did not use animation, I only used that here as a way to reproduce the bug, but this happens if you change the shape "extents" property in any way.

**Comment by Salabasama** [comment]
I seem to be experiencing this bug now, in 3.2.3.  My area appears to stop being able to detect any area after that area's shape extents are changed.

**Comment by akien-mga** [comment]
@Salabasama Please test with 3.3 RC 6, and if still reproducible, open a new issue with a minimal reproduction project.

**Comment by Salabasama** [comment]
> @Salabasama Please test with 3.3 RC 6, and if still reproducible, open a new issue with a minimal reproduction project.

I am not experiencing this issue in 3.3 RC 6.  Thank you.
